### PR TITLE
Exclude BCL binaries in Roslyn ServiceHub vsix

### PR DESCRIPTION
**Pull request by genlu**
VS moved to .NET 8 in [this](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/477867) so we no longer need to ship BCL binaries in our own vsix.

[val buid](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=8150691&view=results)

[test insertion](https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/487208)

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets [hunk 1]
```diff
@@ -145,7 +145,7 @@
         two would eventually be in sync. But we'd need to include those binaries during the out-of-sync period.
         Note, for the same reasone, we can't safely exclude shared dependencies from ServiceHub host folder.
```
Comment (jasonmalinowski) [inline src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets:146]: Does this comment need to be updated?

File: src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets [hunk 2]
```diff
@@ -145,7 +145,7 @@
...
       -->
-      <_PublishRuntimeLibraries>true</_PublishRuntimeLibraries>
+      <_PublishRuntimeLibraries>false</_PublishRuntimeLibraries>
```
Comment (sharwell) [inline src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets:148]: 💡 Can you verify correctness of the preceding comment?

Comment (genlu) [reply src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets:148]: Yes, from the RPS log, I confirmed the loading of binaries like S.C.I and S.R.M are now loaded from the .NET 8 runtime folder in VS

**Comment by genlu** [comment]
Can't merge until integration CI moves to 17.8p1.

**Comment by genlu** [comment]
This should be good to merge now, since main branch integration CI is now on17.8p1. FYI @dibarbet

**Comment by dibarbet** [comment]
@genlu main is not on 17.8 p1 yet.  Only the scouting queue is and integration CI is broken there.

**Comment by genlu** [comment]
Oops, my bad. Make this a draft again

File: src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets [hunk 3]
```diff
@@ -158,8 +158,8 @@
 
       <!-- the only assembly we need under runtime folder (runtimes\win-x64\native\e_sqlite3.dll) is handled by the vsix project directly -->
       <_ExcludedFiles Include="$(PublishDir)runtimes\**\*.*" />
-
-      <_ExcludedFiles Condition="'$(_PublishRuntimeLibraries)' == 'false'" Include="@(_RuntimeLibrariesInPublishDir)" />
+      <!-- Always include BCL in the vsix for CI build to avoid breakage. Since VS used on CI machines usually lags behind our insertion target -->
+      <_ExcludedFiles Condition="'$(_PublishRuntimeLibraries)' == 'false' and '$(ContinuousIntegrationBuild)' != 'true'" Include="@(_RuntimeLibrariesInPublishDir)" />
```
Comment (genlu) [inline src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets:162]: I have merged the change after this, which is similar to how we use `Roslyn.VisualStudio.Setup.Dependencies`

**Pull request merged**